homeassistant/components/vesync/entity.py

```diff
@@ -6,17 +6,24 @@
 
 from homeassistant.helpers.device_registry import DeviceInfo
 from homeassistant.helpers.entity import Entity, ToggleEntity
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
 
 from .const import DOMAIN
 
 
-class VeSyncBaseEntity(Entity):
+class VeSyncBaseEntity(CoordinatorEntity, Entity):
```

Comment:
```suggestion
class VeSyncBaseEntity(CoordinatorEntity[..]):
```
Add the coordinator type to the typeslot of the coordinator entity